cookbooks/stac-zarr-report/data-producers/index.qmd

```diff
@@ -0,0 +1,184 @@
+- Data variables are not exposed at the STAC level, so users cannot discover relevant datasets by searching for variables.
+- User is responsible for aligning and concatenating data which can be slow and error-prone.
+
+### Many smaller Zarr stores - some STAC medatata
+
+Same as above but with the additional step:
+
+5. Write metadata on the STAC item using the data cube extension.
+
+**Pros**
+- Data variables are exposed at the STAC level, so users can discover relevant datasets by searching for variables.
+- Spatial temporal extents are exposed at the STAC level, so users can filter and subset data without accessing the data files themselves.
+- When only accessing necessary items, fewer GETs to construct the data cube.
+
+**Cons**
+- Metadata is fully duplicated which can be expensive to maintain and potentially lead to inconsistencies if the underlying data is changed
+- User is responsible for aligning and concatenating data which can be slow and error-prone.
+
+### Virtual dataset in an external file - minimal STAC metadata
+
+This is similar to **One big Zarr store - minimal STAC metadata** with the exception that data does not need to be stored in a Zarr store. It can be stored in any number of COG or NetCDF or HDF5 files. Anything that has consistent chunking, encoding, and aligned coordinates on-disc.
+
+1. Create a virtual reference file (kerchunk) or store (icechunk) pointing to chunks of data on-disc.
+2. Define a STAC collection for each virtual dataset.
+3. In each collection include a collection-level asset containing the href link to the virtual reference file.
+
+**Pros**
+- There is no metadata duplication in STAC, so the STAC side is easy to maintain.
+- Simple access interface for Python users - no client-side concatenation.
+- Entire data cube can be lazily constructed with one GET to the reference file or store.
+
+**Cons**
+- Data variables are not exposed at the STAC level, so users cannot discover relevant datasets by searching for variables.
+- Virtual reference file needs to be kept in sync with updates in the underlying data which can be expensive to maintain.
+- Non-Python access is less well developed.
+
+### Virtual dataset in an external file - more STAC metadata
+
+Same as above but with the additional step:
+
+4. Write STAC item metadata using the data cube extension.
+
+**Pros**
+- Data variables are exposed at the STAC level, so users can discover relevant datasets by searching for variables.
+- Simple access interface for Python users - no client-side concatenation.
+- Entire data cube can be lazily constructed with one GET to the reference file.
+
+**Cons**
+- Metadata is duplicated, but it is variable-level so unlikely to change often.
+- Virtual reference file needs to be kept in-sync with updates in the underlying data which can be expensive to maintain.
+- Non-Python access is less well developed.
+
+### Virtual dataset in STAC - most STAC metadata
```

Comment:
Generally, I like it. This helps bridge the gap between searchability and data access